full-service/src/service/sync.rs

```diff
@@ -91,44 +86,51 @@ pub struct SyncThread {
     /// The main sync thread handle.
     join_handle: Option<thread::JoinHandle<()>>,
 
-    /// Stop trigger, used to signal the thread to reminate.
+    /// Stop trigger, used to signal the thread to terminate.
     stop_requested: Arc<AtomicBool>,
 }
 
 impl SyncThread {
+    /// Dispatches account sync tasks to workers.
+    ///
+    /// # Arguments
+    /// * `ledger_db` - Local instance of the MobileCoin ledger.
+    /// * `wallet_db` - The wallet database.
+    /// * `num_workers` - Number of worker threads. Defaults to num_cpus.
+    /// * `logger` - Logger.
     pub fn start(
         ledger_db: LedgerDB,
         wallet_db: WalletDb,
         num_workers: Option<usize>,
         logger: Logger,
     ) -> Self {
-        // Queue for sending jobs to our worker threads.
+        // Worker task queue.
         let (sender, receiver) = crossbeam_channel::unbounded::<SyncMsg>();
 
-        // A hashset to keep track of which AccountIds were already sent to the queue,
-        // preventing them from being sent again until they are processed.
+        // AccountIds that have been added to the queue, but have not yet been
+        // processed.
         let queued_account_ids = Arc::new(Mutex::new(HashSet::<AccountId>::default()));
```

Comment:
Gut feel: this stuff would be easier if the "task master" periodically reads the accounts and the ledger state, and then dispatches finite tasks like "Apply account X to blocks Y..Z", instead of letting the worker thread figure out what work needs to be done.